daemon/containerd/image.go

```diff
@@ -44,19 +44,33 @@ func (i *ImageService) GetImage(ctx context.Context, refOrID string, options ima
 	}
 
 	cs := i.client.ContentStore()
-	conf, err := containerdimages.Config(ctx, cs, desc, platform)
-	if err != nil {
-		return nil, err
-	}
 
-	imageConfigBytes, err := content.ReadBlob(ctx, cs, conf)
+	var manifests []ocispec.Manifest
+	err = i.walkImageManifests(ctx, desc, func(img *ImageManifest) error {
+		mfst, err := img.Manifest(ctx)
+		if err != nil {
+			return err
+		}
+		manifests = append(manifests, mfst)
+		return nil
+	})
 	if err != nil {
 		return nil, err
 	}
 
 	var ociimage ocispec.Image
-	if err := json.Unmarshal(imageConfigBytes, &ociimage); err != nil {
-		return nil, err
+	for _, mfst := range manifests {
+		imageConfigBytes, err := content.ReadBlob(ctx, cs, mfst.Config)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := json.Unmarshal(imageConfigBytes, &ociimage); err != nil {
+			return nil, err
+		}
+		if platform.Match(ociimage.Platform) {
```

Comment:
This will always be true if no platform is provided:

`platform := platforms.AllPlatformsWithPreference(cplatforms.Default())`
will match all platform (`Match` will be true for any argument).
In this context, we should use its `Less` method to get an ordered slice of the platforms, and then pick the first one.

Reply:
It might be simpler to just move the Config picking inside the `walkImageManifests`:
```go
defaultPlatformMatcher = platforms.Default()
var ociimage ocispec.Image
err := i.walkImageManifests(ctx, desc, func(img *ImageManifest) error {
  cfg, err := img.Config(ctx)
  if err != nil {
     return err
  }
  ociimage = cfg
  if defaultPlatformMatcher.Match(cfg.Platform) {
     break
  }
})
```

Reply:
> It might be simpler to just move the Config picking inside the `walkImageManifests`:

Can't do that afaict since that's not really a loop, but I removed the second loop and instead sort the slice of images and picked the first one as per your suggestion. Lmk if that sounds good or if you have a better idea of how to handle that.